sherpa/optmethods/opt.py

```diff
@@ -33,19 +34,55 @@
            'SimplexStep')
 
 
-P = ParamSpec("P")
+FUNC_MAX = float(np.finfo(np.float64).max)
+
+# Some code assumes the first argument is an ndarray, but is this always
+# the case? For now assume that the data is always sent as an ndarray.
+#
+# sherpa.stats.StatCallback converts a StatFunc into an OptimizerFunc.
+#
+OptimizerFunc = Callable[[np.ndarray], float]
+
+# WorkerFunc could be done as a Protocol, but the argument names are
+# not consistent, and it also requires **kwargs, which has not been
+# handled well by pyright. Concatenate could be used to at least
+# indicate the extra arguments but that requires Python 3.11 (as
+# Concatenate in Python 3.10 can not end with an ellipsis).
+#
+MyOptOutput = tuple[int, float, np.ndarray]
```

Comment:
So, here is where `MyOptOutput` is named. We have three issues

- essentially no documentation
- should it be more structured - e.g. a `dataclass` - so that data access is more obvious than just indexing into a container
- the name sucks

Reply:
- name suckage

I don't have a strong feeling about this. I thought some of the existing code has `MyOpt` in it, which is why I came up with the name, but now I see no evidence of this. It was probably `MyNCores` and then thinking it's `OptOutput` and I merged it together. So I'll likely rename it to `OptOutput`.

Reply:
So, I've now 

- changed it to `OptOutput`
- added some documentation
- then changed the `tuple` to a `dataclass`

Whether the dataclass step is worth it is up for discussion.

Reply:
One advantage of this approach is that it is future-proof. If we add an additional element to `OptOutput` in the future, then a dataclass like this can be changed without problems because everything is accessed by name. Scipy uses the same approach in e.g. optimization they return an object and you use `out.nfev`, `out.root` etc. In later versions you can then add `out.extra_message` without backwards-compatibility concerns.